extensions/sql-bindings/src/dialogs/addSqlBindingQuickpick.ts

```diff
@@ -45,60 +48,80 @@ export async function launchAddSqlBindingQuickpick(uri: vscode.Uri | undefined):
 
-	// 3. ask for object name for the binding
-	const objectName = await azureFunctionsUtils.promptForObjectName(selectedBinding.type);
+		// 3. ask for object name for the binding
+		quickPickStep = 'getObjectName';
+		const objectName = await azureFunctionsUtils.promptForObjectName(selectedBinding.type);
 
-	if (!objectName) {
-		return;
-	}
+		if (!objectName) {
+			return;
+		}
+		TelemetryReporter.createActionEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.getObjectName)
+			.withAdditionalProperties(propertyBag).send();
+
+		// 4. ask for connection string setting name
+		let projectUri: vscode.Uri | undefined;
+		try {
+			projectUri = await azureFunctionsUtils.getAFProjectContainingFile(uri);
+		} catch (e) {
+			// continue even if there's no AF project found. The binding should still be able to be added as long as there was an azure function found in the file earlier
+		}
 
-	// 4. ask for connection string setting name
-	let projectUri: vscode.Uri | undefined;
-	try {
-		projectUri = await azureFunctionsUtils.getAFProjectContainingFile(uri);
-	} catch (e) {
-		// continue even if there's no AF project found. The binding should still be able to be added as long as there was an azure function found in the file earlier
-	}
+		quickPickStep = 'updateConnectionString';
+		let connectionStringSettingName = await azureFunctionsUtils.promptAndUpdateConnectionStringSetting(projectUri);
+		if (!connectionStringSettingName) {
+			return;
+		}
+		TelemetryReporter.createActionEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.updateConnectionString)
+			.withAdditionalProperties(propertyBag).send();
 
-	let connectionStringSettingName = await azureFunctionsUtils.promptAndUpdateConnectionStringSetting(projectUri);
-	if (!connectionStringSettingName) {
-		return;
-	}
+		// 5. insert binding
+		try {
+			const result = await addSqlBinding(selectedBinding.type, uri.fsPath, azureFunctionName, objectName, connectionStringSettingName);
 
-	// 5. insert binding
-	try {
-		const result = await addSqlBinding(selectedBinding.type, uri.fsPath, azureFunctionName, objectName, connectionStringSettingName);
+			if (!result.success) {
+				void vscode.window.showErrorMessage(result.errorMessage);
+				TelemetryReporter.createErrorEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.finishAddSqlBinding)
+					.withAdditionalProperties(propertyBag).send();
+				return;
+			}
+			TelemetryReporter.createActionEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.finishAddSqlBinding)
+				.withAdditionalProperties(propertyBag).send();
 
-		if (!result.success) {
-			void vscode.window.showErrorMessage(result.errorMessage);
-			TelemetryReporter.sendErrorEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.finishAddSqlBinding);
+		} catch (e) {
+			void vscode.window.showErrorMessage(utils.getErrorMessage(e));
+			TelemetryReporter.createErrorEvent(TelemetryViews.SqlBindingsQuickPick, TelemetryActions.finishAddSqlBinding)
```

Comment:
Suggest adding errorType here as well (even if we don't get a timeout error, that way if we improve getErrorType later on to include more specific types of errors we don't have to remember to go update this one)